src/pybind/mgr/dashboard/frontend/src/app/shared/components/usage-bar/usage-bar.component.ts

```diff
@@ -24,30 +25,85 @@ export class UsageBarComponent implements OnChanges {
   calculatePerc = true;
   @Input()
   title = $localize`usage`;
-  @Input()
-  customLegend?: string;
-  @Input()
-  customLegendValue?: string;
-  @Input()
-  showFreeToolTip = true;
-  @Input()
-  showMultisiteTooltip = false;
 
-  usedPercentage: number;
-  freePercentage: number;
+  data: any;
+  options: any;
+  usedPercentage: number = 0;
+  freePercentage: number = 0;
+
+  constructor(private dimless: DimlessPipe, private dimlessBinary: DimlessBinaryPipe) {
+    this.data = [
+      {
+        group: 'Capacity',
+        value: 0
+      }
+    ];
+    this.options = {
+      resizeable: false,
+      title: this.title,
+      meter: {
+        peak: 100,
+        showLabels: false,
+        height: '10px',
+        width: '100px'
+      },
+      tooltip: {
+        enabled: true
+      },
+      color: {
+        scale: {
+          Capacity: '#0043ce'
+        }
+      },
+      height: '80px',
+      width: '200px',
+      toolbar: {
+        enabled: false
+      }
+    };
+  }
 
   ngOnChanges() {
-    if (this.calculatePerc) {
-      this.usedPercentage = this.total > 0 ? (this.used / this.total) * 100 : 0;
-      this.freePercentage = 100 - this.usedPercentage;
+    this.usedPercentage = this.total > 0 ? (this.used / this.total) * 100 : 0;
+    this.freePercentage = 100 - this.usedPercentage;
+    this.data[0].value = Math.round(this.total > 0 ? (this.used / this.total) * 100 : 0);
+    if (this.isBinary) {
+      this.options.title = `${this.dimlessBinary.transform(
+        this.used
+      )} / ${this.dimlessBinary.transform(this.total)}`;
     } else {
-      if (this.used) {
-        this.used = this.used.slice(0, -1);
-        this.usedPercentage = Number(this.used);
-        this.freePercentage = 100 - this.usedPercentage;
-      } else {
-        this.usedPercentage = 0;
-      }
+      this.options.title = `${this.dimless.transform(this.used)} / ${this.dimless.transform(
+        this.total
+      )}`;
+    }
+
+    if (
+      this.warningThreshold &&
+      this.warningThreshold >= 0 &&
+      this.usedPercentage / 100 >= this.warningThreshold
+    ) {
+      this.options = {
+        ...this.options,
+        color: {
+          scale: {
+            Capacity: '#f1c21b'
+          }
+        }
+      };
+    }
+    if (
+      this.errorThreshold &&
+      this.errorThreshold >= 0 &&
+      this.usedPercentage / 100 >= this.errorThreshold
+    ) {
+      this.options = {
+        ...this.options,
+        color: {
+          scale: {
```

Comment:
90% of the times we should never touch the OnChanges hook. Doing that usually ends up complicating the change detection mechanism.

Also I see it's been used for:
* Data processing (percentage),
* Data formatting (units),
* Data presentation (colouring).

All those services should be done elsewhere:
* Processing -> Services (eiter front-end and, when possible, back-end, so that we have ready-to-display info from the back-end and minimize data processing in the browser),
* Formatting and Presentations should always happen in the HTML template, through Pipes, HTML, CSS...

We should always try to create and consume components in an HTML template driven way:
```html
<cd-chart-usage
  *ngIf="capacityService$ | async as capacity"
  [used]="capacity.used"
  [total]="capacity.total"
  [lowThreshold]="capacity.warningThreshold"
  [highThreshold]="capacity.errorThreshold"
></cd-chart-usage>

And then, our wrapper `cd-chart-usage` component will do things like:
```html
<ibm-meter-chart
  [data]="[{group: $localize`Used`, value: used}]"
  [options]="{
    title: $localize`Capacity Usage`,
    width: "<relative_unit_width",
    ...
  }"
></ibm-meter-chart>
```

```
This means:
* Using `AsyncPipe` for subscribing to service observables directly from the template, not from the component logic (.ts file).
* Using Pipes and `$localize` to transform data.
* Relying on UI framework (CSS/SCSS) variables rather than hardcoding hex numbers or pixel widths.